index.js

```diff
@@ -355,12 +355,16 @@ AFRAME.registerShader('html', {
     this.__ctx.drawImage(canvas, 0, 0, cnvW, cnvH)
     this.__texture.needsUpdate = true
     if (this.__ratio) {
-      /* change size */
-      const { width, height } = this.el.getObject3D('mesh').geometry.parameters
-      this.el.setAttribute('geometry', Object.assign({}, this.el.getAttribute('geometry'), {
-        width: (this.__ratio === 'width')? width : height * ratio,
-        height: (this.__ratio === 'width')? width / ratio : height
-      }))
+      if (this.el.getObject3D('mesh').geometry.metadata) {
+        var _el$getObject3D$geome = this.el.getObject3D('mesh').geometry.metadata.parameters;
```

Comment:
I almost forgot the pr! but where I use it is hover show/hide event, so when I unmount the element, it shows undefined error in console